test/run/run-e2e-tests.js

```diff
@@ -0,0 +1,94 @@
+#!/usr/bin/env node
+
+const path = require( 'path' );
+const { promisify } = require( 'util' );
+const { execSync, spawn } = require( 'child_process' );
+const { existsSync, openSync, mkdirSync } = require( 'fs' );
+
+const PROJECT_DIR = path.join( __dirname, '../../' );
+const BUILT_APP_DIR = path.join( PROJECT_DIR, 'release', 'mac', 'WordPress.com.app', 'Contents', 'MacOS' );
+
+function spawnDetached( cwd, command, args, output ) {
+    const app = spawn( command, args, { stdio: [ 'ignore', output, output ], detached: true, cwd } );
+    app.on( 'error', err => {
+        throw `failed to initialize command "${ command }": "${ err }"`;
+    } );
+    return app;
+}
+
+function initLogs( timestamp ) {
+    const dir = path.join( PROJECT_DIR, 'test', 'logs', `${ timestamp }` );
+
+    if ( !existsSync( dir ) ) {
```

Comment:
note that this introduces a race condition that may not protect us. the directory could be created between the call to `exists` and the call to `mkdir`

probably simpler to just call `mkdirSync` and handle failure if the directory already exists. it may only fail in windows.

in other words, for the same level of safety we can skip `existsSync()`

```js
try {
	mkdirSync( dir, { recursive: true } );
} catch (e) {
	// pass
}
```

or do the more elegant/complete thing

```js
try {
	mkdirSync( dir, { recursive: true } );
} catch (e) {
	if (!existsSync(dir)) {
		console.error('Could not create log directory');
		process.exit(1);
	}
}
```

Reply:
According to the docs it may never throw an error if `recursive: true` and we're not creating `/` on Windows

> Calling fs.mkdir() when path is a directory that exists results in an error only when recursive is false.

so in that case we may have the same safety as with `existsSync()` without any further try/catch. if it fails the build crashes as it must.

```js
…

mkdirSync(dir, {recursive: true });

…
```

Reply:
Thanks for double-checking the usage here. Simplified this in f4021b9